warehouse/ingest-core/src/main/java/datawave/ingest/util/GenerateShardSplits.java

```diff
@@ -186,4 +287,43 @@ public static void main(String[] args) throws Exception {
 
         }
     }
+
+    private static Set<String> getTabletLocations(AccumuloClient client, String tableName, List<Range> rangesToWaitFor)
+                    throws AccumuloException, AccumuloSecurityException, TableNotFoundException {
+        Locations locations = client.tableOperations().locate(tableName, rangesToWaitFor);
+        Set<String> tabletLocations = new HashSet<>();
+        for (Range range : rangesToWaitFor) {
+            TabletId id = locations.groupByRange().get(range).get(0);
+            tabletLocations.add(locations.getTabletLocation(id));
+            System.out.println(range.getStartKey() + "," + range.getEndKey() + "," + locations.getTabletLocation(id));
+        }
+        return tabletLocations;
+    }
+
+    protected static List<Text> calculateMidpoints(List<Text> splits) {
```

Comment:
Could potentially pass down the output list and keep adding to it.

```suggestion
    protected static void calculateMidpoints(List<Text> splits, List<Text> midpoints) {
```